asmcomp/cmmgen.ml

```diff
@@ -2419,9 +2423,10 @@ and transl_letrec env bindings cont =
 (* Translate a function definition *)
 
 let transl_function f =
+  let body = Un_anf.apply f.body ~what:f.label in
```

Comment:
This seems to be the only call to this function which is not protected by a check on flambda mode.  Instead of doing the check inside the function, one could do it here to make it  that nothing changes in non flambda mode.